pkg/nodediscovery/nodediscovery.go

```diff
@@ -16,6 +16,7 @@ package nodediscovery
 
 import (
 	"context"
+	goerrors "errors"
```

Comment:
Nit: I would rename the k8s package instead of the standard library one, but it's a very small not.